packages/core/src/tracing/cls.js

```diff
@@ -549,31 +552,50 @@ function skipExitTracing(options) {
       isActive: true,
       extendedResponse: false,
       skipParentSpanCheck: false,
-      skipIsTracing: false
+      skipIsTracing: false,
+      checkReducedSpan: false
     },
     options
   );
 
-  const parentSpan = getCurrentSpan();
+  let parentSpan = getCurrentSpan();
+
+  // If there is no active entry span, we fall back to the reduced span of the most recent entry span.
+  // See comment in packages/core/src/tracing/clsHooked/unset.js#storeReducedSpan.
+  if (opts.checkReducedSpan) {
+    parentSpan = getCurrentSpan() || getReducedSpan();
+  }
+
   const suppressed = tracingSuppressed();
   const isExitSpanResult = isExitSpan(parentSpan);
 
   // CASE: first ask for suppressed, because if we skip the entry span, we won't have a parentSpan
   //       on the exit span, which would create noisy log messages.
   if (suppressed) {
-    if (opts.extendedResponse) return { skip: true, suppressed, isExitSpan: isExitSpanResult };
+    if (opts.extendedResponse) return { skip: true, suppressed, isExitSpan: isExitSpanResult, parentSpan };
     return true;
   }
 
+  if (allowRootExitSpan) {
+    if (opts.extendedResponse) return { skip: false, suppressed, isExitSpan: isExitSpanResult, parentSpan };
+    else return false;
+  }
+
+  if (opts.checkReducedSpan && (!parentSpan || isExitSpan(parentSpan))) {
+    if (opts.extendedResponse) return { skip: true, suppressed, isExitSpan: isExitSpanResult, parentSpan };
+    else return true;
+  }
+
   if (!opts.skipParentSpanCheck && (!parentSpan || isExitSpanResult)) {
-    if (opts.extendedResponse) return { skip: true, suppressed, isExitSpan: isExitSpanResult };
+    if (opts.extendedResponse) return { skip: true, suppressed, isExitSpan: isExitSpanResult, parentSpan };
     else return true;
   }
 
   const skipIsActive = opts.isActive === false;
   const skipIsTracing = !opts.skipIsTracing ? !isTracing() : false;
   const skip = skipIsActive || skipIsTracing;
-  if (opts.extendedResponse) return { skip, suppressed, isExitSpan: isExitSpanResult };
+
+  if (opts.extendedResponse) return { skip, suppressed, isExitSpan: isExitSpanResult, parentSpan };
```

Comment:
Putting it here.

In bull.js we can use the returned parentSpan.

>     const parentSpan = cls.getCurrentSpan();
>    const skipTracingResult = cls.skipExitTracing({

Reply:
Same for ampq.js

>   const skipTracingResult = cls.skipExitTracing({ isActive, extendedResponse: true, skipParentSpanCheck: true });
>  const parentSpan = cls.getCurrentSpan();
>  const isExitSpan = skipTracingResult.isExitSpan;

Reply:
As well as: redis.js, ioredis.js, not sure about db2 🤔

Reply:
redis and ioredis does not use the extendedResponse attribute and thus the result from skipExitTracing function will be a boolean. So can we leave it ?

In db2, intead of falling back to reducedSpan     `const parentSpan = cls.getCurrentSpan() || prepareCallParentSpan;
` prepareCallParentSpan is used. So the reducedSpan logic may not work here

Reply:
Three cases:
1. no parent span
2. entry span
3. exitspan

Idea: add this as an integration test.
app_mixed.js

worker.js
```
instana = {allowNoEntrySpan:true}

// trace 1
async ioredis.send 

// trace 2
async instana.async.sdk.startEntry
async ioredis.send
```